src/interface/obsidian/src/modal.ts

```diff
@@ -90,11 +90,42 @@ export class KhojModal extends SuggestModal<SearchResult> {
     }
 
     async renderSuggestion(result: SearchResult, el: HTMLElement) {
+        // Max Size (words or number of lines)
+        let lines_to_render = 8;
         let words_to_render = 30;
-        let entry_words = result.entry.split(' ')
-        let entry_snipped_indicator = entry_words.length > words_to_render ? ' **...**' : '';
-        let snipped_entry = entry_words.slice(0, words_to_render).join(' ');
-        MarkdownRenderer.renderMarkdown(snipped_entry + entry_snipped_indicator, el, null, null);
+        var path = "";
+        var basename = "";
+
+        // set path and basename of file depending on OS
+        if (result.file.indexOf('\\') > -1) {
+            path = result.file.split('\\').slice(0, -1).join('\\');
+            basename = result.file.split('\\').slice(-1)[0];
+        }
+        if (result.file.indexOf('/') > -1) {
+            path = result.file.split('/').slice(0, -1).join('/');
+            basename = result.file.split('/').slice(-1)[0];
+        }
+        basename = basename.replace('.md', '');
+        let entry_lines_split = result.entry.split('\n')
+        // remove frontmatter if present
+        if (entry_lines_split[0] == '# ---'){
+            //loop through lines until we find the end of the frontmatter
+            for (let i = 1; i < entry_lines_split.length; i++){
+                if (entry_lines_split[i] == '---'){
+                    //remove the frontmatter
+                    entry_lines_split = entry_lines_split.slice(i+1);
+                    break;
+                }
+            }
+        }
+        entry_lines_split = entry_lines_split.join('\n').trim().split('\n');
+        let entry_lines = entry_lines_split.slice(0, lines_to_render).join('\n');
+        let entry_words_split = entry_lines.split(' ');
+        let entry_snipped_indicator = entry_words_split.length > words_to_render ? ' **...**' : '';
+        let entry_words = entry_words_split.slice(0, words_to_render).join(' ').trim();
+        el.createEl("div",{ cls: 'khoj-result-file' }).setText(basename);
+        var result_div = el.createEl("div", { cls: 'khoj-result-entry' })
+        MarkdownRenderer.renderMarkdown(entry_words + entry_snipped_indicator, result_div, path, null);
```

Comment:
1. The `renderMarkdown` expects the markdown file path. This seems to pass the path to the markdown file's directory instead. So passing `results.file` maybe more relevant than passing `path`?
2. I'd left this as `null` as (relative/internal) links within a result shown in the search modal can't be interacted with anyway.

Let me know if you know of any benefits to passing the file path in to the renderer? Else IMO, we should just pass `null` for the `path` argument